bounceview/src/main/java/com/github/hariprasanths/bounceview/BounceView.java

```diff
@@ -345,34 +298,37 @@ private void setAnimToTabList() {
         if (mTabList.get() != null) {
             for (int i = 0; i < mTabList.get().getTabCount(); i++) {
                 final TabList.Tab tab = mTabList.get().getTabAt(i);
-                tab.setTouchEventListener((component, motionEvent) -> {
-                    int action = motionEvent.getAction();
-                    if (action == TouchEvent.PRIMARY_POINT_DOWN) {
-                        isTouchInsideView = true;
-                        startAnimScale(component, pushInScaleX, pushInScaleY,
-                                pushInAnimDuration, pushInInterpolator, 0);
-                        return true;
-
-                    } else if (action == TouchEvent.PRIMARY_POINT_UP) {
-                        if (isTouchInsideView) {
-                            component.createAnimatorProperty().cancel();
-                            startAnimScale(component, popOutScaleX, popOutScaleY, popOutAnimDuration,
-                                    popOutInterpolator, 0);
-                            startAnimScale(component, 1f, 1f, popOutAnimDuration,
-                                    popOutInterpolator, popOutAnimDuration + 1);
-                            tab.select();
-                            return false;
-                        }
-                    } else {
-                        return checkTabActions(action, component, motionEvent);
-                    }
-                    return false;
-                });
+                check(tab);
             }
         }
     }
 
-    private boolean checkTabActions(int action, Component component, TouchEvent motionEvent) {
+    private void check(Component component1) {
+        component1.setTouchEventListener((component, motionEvent) -> {
+            int action = motionEvent.getAction();
+            if (action == TouchEvent.PRIMARY_POINT_DOWN) {
+                isTouchInsideView = true;
+                startAnimScale(component, pushInScaleX, pushInScaleY,
+                        pushInAnimDuration, pushInInterpolator, 0);
+                return true;
+
+            } else if (action == TouchEvent.PRIMARY_POINT_UP) {
+                if (isTouchInsideView) {
+                    component.createAnimatorProperty().cancel();
+                    startAnimScale(component, popOutScaleX, popOutScaleY, popOutAnimDuration,
+                            popOutInterpolator, 0);
+                    startAnimScale(component, 1f, 1f, popOutAnimDuration,
+                            popOutInterpolator, popOutAnimDuration + 1);
+                    return false;
+                }
+            } else {
+                return checkActions(action, component, motionEvent);
+            }
+            return false;
+        });
+    }
+
+    private boolean checkActions(int action, Component component, TouchEvent motionEvent) {
```

Comment:
rename the function as "setTouchEventActionsInternal"

Reply:
done